### ISSUE TITLE
Stuck during training

### ISSUE THREAD

**Issue by 646503029** [JingyiXu404/TPAMI-DeepM2CDL#5]
Could you please tell me why is it that during the training of my denoising model, the network does not continue to converge after a period of training and the loss no longer decreases. The dataset and code I am using has not been altered in any way.

**Comment by luckyJingyi** [comment]
Hello, I'm Jingyixu. During my previous training, When I increased the number of layers or the scales, I did encounter training issues, and I needed to correspondingly reduce the learning rate or perform layer-by-layer training. However, I haven't encountered the situation you described. Could you elaborate on the problem in more detail? For example, to what extent has the learning rate reached? What does the corresponding output image look like? And what is the PSNR value?

**Comment by 646503029** [comment]
Thank you for your attention and response to my question! First of all, let me add the specifics of the problem: I noticed an unusual stagnation of the evaluation metrics while training the full process using a network with noise level σ = 50. The specific training log shows that the PSNR value of the validation set shows a steady upward trend in the first 700 iteration cycles, but from the 900th iteration onwards, the metric is permanently fixed at 27.41. Even after continuous training up to the 7100th iteration cycle, there is still no fluctuating change in the value, and eventually I have to terminate the training process voluntarily.

Is this the expected convergence behaviour of the model? Or is there a hyperparameter that needs to be adjusted during the training process?


![Image](https://github.com/user-attachments/assets/0e2ae8a6-4a4e-44b3-9526-799763786de4)

![Image](https://github.com/user-attachments/assets/b6a80e31-2597-4222-8ab2-5e0d994e52f0)

![Image](https://github.com/user-attachments/assets/1cf494ce-f32d-467f-858c-ba02ebdf4b21)

**Comment by luckyJingyi** [comment]
According to your picture, the model achieved the best PSNR=27.41 on the test set at epoch=16, the subsequent rounds did not exceed this value, which does not mean convergence. You can try to adjust the learning rate from 1e-4 to 5e-5 and continue training based on this best model in epoch 16.

**Comment by 646503029** [comment]
Thank you very much for your suggestion, I will try to adjust the learning rate to 5e-5 and resume training from round 16 of the model.
